### DISCUSSION TITLE
[FR]  Option to use Stun as Rendezvous protocol and possible Turn as Relay protocol

### DISCUSSION THREAD

**Discussion by oiaohm** [rustdesk/rustdesk#3202] [General]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Both Stun  RFC 5389 and Turn RFC 8656/7065 are already defined standards used for the Rendezvous and Relay by many other protocols. 

Main use cases would be increased resilience of rustdesk due to being able to use more existing servers.  Also possible increase performance by users having more server options so able to choose closer servers in some cases.

I don't see Rendezvous being highly performance critical but relays absolutely are.   Every extra relays you can get the better.

### Impact

Due to many parties having Stun servers on the internet for free like google this could increase rustdesk resilience.   Like if the current 6 public servers of the rustdesk Rendezvous go off line functionality would be bricked but means to fall back and use Stun would still allow functionality to keep on going.   Also due to more Stun servers existing than rustdesk servers is more likely for one to be closer and faster.

There 200+ active public stun servers at any one time this well passes the public 6 rust.  Just adding google 4 stun servers alone would be a massive increase in resilience.   There are open source projects that run their own stun servers like stunprotocol and others there is would be around another 10 servers at least.

Turn servers are rare provided publicly but they do exist this is about 6 most in very different locations to the current 6 provided by rustdesk.

Without digging deeper I don't know if the rustdesk  Rendezvous and relay protocols are providing any specialist advantage over using Stun and Turn other than restricting clients that can connect to being rustdesk.

### Additional Context

https://github.com/miroslavpejic85/p2p/
Above is a remote desktop client done in C# using stun for Rendezvous side of the connection so it possible to Stun for this without question.

**Comment by rustdesk** [comment]
PR is welcome.

**Comment by rustdesk** [comment]
Also added this to https://github.com/rustdesk/rustdesk/wiki/Google-Summer-of-Code-2023-Ideas#stun-protocol-support
